.github/workflows/stm32build.yml

```diff
@@ -78,47 +78,90 @@ jobs:
     strategy:
       matrix:
         # Parse JSON array containing names of all filters matching any of changed files
         package: ${{ fromJSON(needs.changes.outputs.packages) }}
 
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3                         # gets file environment with submodule included for libraries
-      with:
-        submodules: 'true'
-    - uses: carlosperate/arm-none-eabi-gcc-action@v1    # Setup the compiler and compile the program
-    - run: make
-      working-directory: 'STM32/${{ matrix.package }}/'
-    - uses: actions/upload-artifact@v3                  # Upload the .elf and .bin file under the summary tab
-      with: 
-        name: ${{ matrix.package }}
-        path: | 
-          STM32/${{ matrix.package }}/build/${{ matrix.package }}.elf
-          STM32/${{ matrix.package }}/build/${{ matrix.package }}.bin
-    - uses: copenhagenatomics/mono_release_on_push_action@master
-      id: release-on-push
-      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # tag only pushes to the main branch
-      with:
-        bump_version_scheme: minor
-        use_github_release_notes: true
-        tag_prefix: "${{ matrix.package }}-"
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    - name: Read PCB version
-      id: pcbversion
-      run: echo "version=$(head -1 STM32/${{ matrix.package }}/pcbversion)" >> $GITHUB_OUTPUT
-    - name: Release
-      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # upload only pushes to the main branch
-      uses: softprops/action-gh-release@v0.1.15
-      with:
-        files: |
+      # gets file environment with submodule included for libraries
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      # Setup the compiler and compile the program
+      - uses: carlosperate/arm-none-eabi-gcc-action@v1
+      # Generate the next release tag for including in the build. Note: dry_run. It doesn't actually
+      # Release in case something goes wrong in the build
+      - uses: copenhagenatomics/mono_release_on_push_action@master
+        id: release-on-push-1
+        with:
+          bump_version_scheme: minor
+          dry_run: "true"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # Build the upload artifacts
+      - run: make
+        working-directory: "STM32/${{ matrix.package }}/"
+        env:
+          VERSION_TAG: ${{ steps.release-on-push-1.outputs.tag_name }}
+        # Upload the .elf and .bin file under the summary tab
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.package }}
+          path: |
+            STM32/${{ matrix.package }}/build/${{ matrix.package }}.elf
+            STM32/${{ matrix.package }}/build/${{ matrix.package }}.bin
+      - uses: copenhagenatomics/mono_release_on_push_action@master
```

Comment:
What is the reason for calling the mono-release-on-push action? Maybe it makes sense since the first is a dry-run, but don't really understand.

Reply:
In the standard workflow, mono-release-on-push creates a release, and the later "Release" step only appends the bin & elf file to the release. Since the first call of mono-release-on-push doesn't create the release, I thought it best to follow as close as possible the existing workflow (although we could potentially use the "Release" step to create the release AND upload the files instead)

Reply:
As far as I can tell, there are only minor formatting differences between the release created by mono-release-on-push and Release

Reply:
Makes sense :+1: